modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloaderTaskExecutor.java

```diff
@@ -66,6 +68,7 @@ public final class GeoIpDownloaderTaskExecutor extends PersistentTasksExecutor<G
     private final Settings settings;
     private final PersistentTasksService persistentTasksService;
     private final AtomicReference<GeoIpDownloader> currentTask = new AtomicReference<>();
+    private final Semaphore taskSemaphore = new Semaphore(1);
```

Comment:
Would this synchronization logic continue to work if the node that holds this semaphore disappears randomly?